test/test_nn.py

```diff
@@ -314,13 +314,17 @@ class TestNN(NNTestCase):
 
     def _forward(self, module, input):
         with freeze_rng_state():
-            return module(input)
+            if isinstance(input, tuple):
+                return module(*input)
+            else:
+                return module(input)
 
     def _backward(self, module, input, output, grad_output, create_graph=False):
         output.backward(grad_output, retain_graph=True, create_graph=create_graph)
-        if input.grad is None:
-            return None
-        return input.grad.data
+        if isinstance(input, tuple):
+            return tuple(map(lambda i: i.grad.data if i.grad is not None else None, input))
+        else:
+            return input.grad.data if input.grad is not None else None
```

Comment:
Do you think it would be a better API if we just change _forward and _backward such that `input` must always be a tuple? The way it is right now (input can be either Tensor or Tuple) I'm worried that folks will get confused; in addition, the NN module test code is already pretty long and confusing

Reply:
I'm OK with _forward and _backward being able to take `input` as either a Tensor or as a tuple of inputs if we properly type-annotate it with https://github.com/pytorch/pytorch/blob/9dc2bcdc070cd05c2755c6d9b3e9d74a0b1c306c/torch/types.py#L9

Reply:
I have the same feeling about this as check_jacobian one. I think we should keep support both, individual tensor and tuple of tensors. I will update the type-annotate to the `input`.